Goal 1 Stage B: enumeration + JSON contract + dev rebase  Body (copy/paste):

adde401 opened the pull request:
****Summary****

Implements Goal 1 Stage B “grounding”: enumerates concrete forward lines (3 players) and defense pairs (2 players) that satisfy required combo sets.
Enforces canonical uniqueness (no duplicate player_id within a line/pair + symmetry breaking).
Supports multi-combo-per-line/pair activation (0..N combos can trigger on the same line/pair), while keeping “each combo used at most once” at team scope.
Recent updates (to make PR merge-ready)

Rebased as/task15-goal1-stageb onto latest dev and resolved conflicts.
Fix: src/asp/solver.py now imports get_data_loader from src.core.data (previous path caused ModuleNotFoundError).
Stage B JSON output contract: adds schema_version and centralized payload builder (build_stageb_payload) in scripts/goal1_stageb_enumerate.py.
Docs: documents Stage B output contract in docs/GOAL_1.md.
Tests: adds tests/test_goal1_stageb_output_schema.py.
How to test

cd "/Users/sandstrom/NHL 26 Line Combos Optimizer/nhl26-line-combos"
venv/bin/python -m pytest -q
**Test status 102 passed**

adde401 committed:
Goal 1 Stage B: enumeration + ASP core + tests

adde401 committed:
Stage B: tolerate missing optional fields in dataset

andrej-kutny commented on src/asp/rules/full_team.lp:
this file is probably commited by accident, shouldn't be part of G1-B

went quickly through the file and it is too big even for Goal 3 that it is probably trying to solve - we need more smaller files rather than big files - there are some `.lp` files we will pick but not all - it is up to backend to decide which lp files will be used and which will be dynamically created based on target it is working on

also hardcoded part

```lp
min_fwd_line1(87) :- not min_fwd_line1(_).
min_fwd_line2(84) :- not min_fwd_line2(_).
min_def_top4(84)  :- not min_def_top4(_).
min_g1(84)        :- not min_g1(_).

% Require strong first forward line (slots 1-3)
:- min_fwd_line1(M), select(P, S), S=1..3, ovr(P, OVR), OVR < M.

% Require solid second forward line (slots 4-6)
:- min_fwd_line2(M), select(P, S), S=4..6, ovr(P, OVR), OVR < M.
```

is probably only for testing, shouldn't be part of rules

... did not read beyond this part since it is not related to G1-B anyway

andrej-kutny commented on tests/test_solver_solve_any.py:
```diff
@@ -0,0 +1,21 @@
+
+    solver = ASPSolver()
+    if not solver.is_available():
+        pytest.skip("clingo not available in this environment")
```
this should fail the test instead

andrej-kutny commented on tests/test_solver_solve_any.py:
more tests to be added

adde401 replied:
I added an extra Stage B test to cover multi-combo activation + salary bonus summation, and updated Stage B tests to require clingo (no skip). Local: venv/bin/python -m pytest -q → 103 passed. Ready to resolve the ‘more tests’ comment.

andrej-kutny commented on tests/test_base_canon_uniqueness.py:
```diff
@@ -0,0 +1,34 @@
+
+    solver = ASPSolver()
+    if not solver.is_available():
+        pytest.skip("clingo not available in this environment")
```
this should fail

andrej-kutny reviewed (changes requested):
seems like this is trying to achieve too many tasks, it is too big for [#16](https://github.com/andrej-kutny/nhl26-line-combos/issues/16) which is for ASP Goal 1 - Stage B

[#15](https://github.com/andrej-kutny/nhl26-line-combos/issues/15) is for ASP Goal 1 - Stage A

Work should be divided to mulptle tasks and pull requests - Seems like this is trying to achieve [#15](https://github.com/andrej-kutny/nhl26-line-combos/issues/15),  [#16](https://github.com/andrej-kutny/nhl26-line-combos/issues/16) and [#12](https://github.com/andrej-kutny/nhl26-line-combos/issues/12)

If that's the case this needs to be closed and pull requests should be divided to contain only changes regarding the tasks so it is possible to review in shorter time

adde401 committed:
Stage B: add output schema_version contract

adde401 committed:
Fix asp solver import for get_data_loader

adde401 committed:
Scope Stage B: drop full-team rules; require clingo in tests

adde401 committed:
Stage B tests: verify multi-combo activation and require clingo

adde401 committed:
Add Goal 1 Stage B demo pipeline runner

adde401 commented:
PR update:

Added Goal 1 Stage B demo “golden run” generator: scripts/demo_goal1_stageb.py
Added docs: docs/DEMO_PIPELINE.md
Generates (gitignored) demo artifacts: out/demo_goal1_stageb_fwd.json, out/demo_goal1_stageb_def.json
Tests: venv/bin/python -m pytest -q → 103 passed

adde401 commented:
@andrej-kutny I resolved all review conversations and pushed the requested fixes:

Stage B only (removed full-team scope / deleted src/asp/rules/full_team.lp)
clingo-dependent tests now fail instead of skip
added extra Stage B test for multi-combo activation + bonus summation
added demo “golden run” script + docs (scripts/demo_goal1_stageb.py, docs/DEMO_PIPELINE.md)
CI is green (Python 3.11/3.12). Could you please re-review and change your review from “changes requested” to Approve so we can merge to dev?

andrej-kutny commented:
new pr with fixes + extended tests created [Ak/pr24 review (g1b implementation fixes)](https://github.com/andrej-kutny/nhl26-line-combos/pull/24#top)
